nav2_costmap_2d/src/footprint.cpp

```diff
@@ -42,12 +42,14 @@ namespace nav2_costmap_2d
 
 void calculateMinAndMaxDistances(
   const std::vector<geometry_msgs::msg::Point> & footprint,
-  double & min_dist, double & max_dist)
+  double & min_dist_out, double & max_dist_out)
 {
-  min_dist = std::numeric_limits<double>::max();
-  max_dist = 0.0;
+  double min_dist = std::numeric_limits<double>::max();
```

Comment:
Explain, this should also be fine. This is also not going to be fixing any threading problems since there is still no locking. So I doubt this does anything fix-wise

Reply:
before this change, referenced `min_dist_out` aka `LayeredCostmap::inscribed_radius_` is set to an illegal value (std::numeric_limits<double>::max) for a substantial amount of time while the actual radius is being calculated.
any code trying to use `inscribed_radius_`, will get float `inf` instead.
the Obstacle critic was getting NaNs this way.
after this change,` inscribed_radius_` remains a legal value throughout the function and is only changed when the new value is known.

Reply:
But there's still a time period that its being set / potentially gotten during the write. This is not a full solution to a threading problem where one exists. Perhaps instead we should just lock `setFootprint` https://github.com/ros-planning/navigation2/blob/57dd8d4f51e8431dcad419d5da843ba9b0ed67c2/nav2_costmap_2d/src/layered_costmap.cpp#L275C22-L275C34

`  std::unique_lock<Costmap2D::mutex_t> lock(*(combined_costmap_.getMutex()));`

Reply:
the reason I didn't do lock during all of setFootprint() is that the function isn't trivial and takes fair amount of time
better to work on a temp variable and then lock on the assignment.
or simply use atomic assignment.

Reply:
Looking at its use...

`calculateMinAndMaxDistances` called in `LayeredCostmap::setFootprint`, which is called when a new footprint is set on a callback, dynamic parameter, or initialization.  

Dynamic parameter is registered with `this` as is the footprint subscriber, so minus the lifecycle initialization, its mutually exclusive for being `set`. But the map update thread is separate which calls costmap layers that depend on it. 

So... the right answer is to either lock `setFootprint` or change the set semantics for `inscribed_radius_, circumscribed_radius_` to be atomic. But just making the time of the write smaller to only 

```
  min_dist_out = min_dist;
  max_dist_out = max_dist;
```

Isn't a full solution to promise a complete resolution